app/facet_selector/facet_selector.controller.js

```diff
@@ -26,17 +26,22 @@
         return s;
       };
 
-      $scope.set_date_filter = function(start_date, end_date) {
-        if (!start_date && !end_date) {
-          return;
+      var date_is_valid = function(date) {
+        if (date === undefined) {
+          return false;
+        } else {
+          return date.length < 4 ? false : true;
         }
-        start_date = Date.parse(start_date || '0');
-        end_date = Date.parse(end_date || '999999999');
+      }
+
+      $scope.set_date_filter = function(start_date, end_date) {
+        start_date = date_is_valid(start_date) ? Date.parse(start_date) : -62167219200000;
+        end_date = date_is_valid(end_date) ? Date.parse(end_date) : 3093496473600000;
 
-        var facet_fn = function(data) {
+        var facet_fn = function(date) {
           // TODO: handle unparseable dates
           var date_as_int = Date.parse(date);
-          return date_as_int > start_date && date_as_int < end_date;
+          return date_as_int >= start_date && date_as_int <= end_date;
```

Comment:
This doesn't do validation yet; I'm planning to add that to the template in a later PR.